src/MessagingClient.ts

```diff
@@ -214,6 +214,25 @@ export class MessagingClient implements MessagingAPI {
         })
     }
 
+    /**
+     * Listen to updates on the membership of a channel
+     */
+    onChannelJoin(listener: (conversation: Conversation) => void): void {
+        this.matrixClient.on(RoomEvent.MyMembership, (room, membership) => {
+            if (
+                !room || // make sure we have a room
+                room.getType() !== CHANNEL_TYPE || // we only want to know about the updates related to channels
+                membership !== 'join' // we only want to know about the updates related to join a channel
```

Comment:
Would it be better to skip this check and add this information as parameter to the listener? My concern is that we might later need to add more methods to the API based on other channel events.

Reply:
Sure, we can add the information as a parameter 🌱. But keep in mind that we only listen for events related to the membership* update this room received during the sync. So we'd have to add more methods for any other type of channel event anyway.

*membership: join | leave | invite